backend/app/api/threads.py

```diff
@@ -28,59 +29,70 @@ class ThreadMessagesPostRequest(BaseModel):
 
 
 @router.get("/")
-async def list_threads(opengpts_user_id: OpengptsUserId) -> List[Thread]:
+async def list_threads(user: AuthedUser) -> List[Thread]:
     """List all threads for the current user."""
-    return await storage.list_threads(opengpts_user_id)
+    return await storage.list_threads(user["user_id"])
 
 
 @router.get("/{tid}/messages")
 async def get_thread_messages(
-    opengpts_user_id: OpengptsUserId,
+    user: AuthedUser,
     tid: ThreadID,
 ):
     """Get all messages for a thread."""
-    return await storage.get_thread_messages(opengpts_user_id, tid)
+    thread = await storage.get_thread(user["user_id"], tid)
+    if not thread:
+        raise HTTPException(status_code=404, detail="Thread not found")
+    return await storage.get_thread_messages(user["user_id"], thread["thread_id"])
 
 
 @router.post("/{tid}/messages")
 async def add_thread_messages(
-    opengpts_user_id: OpengptsUserId,
+    user: AuthedUser,
     tid: ThreadID,
     payload: ThreadMessagesPostRequest,
 ):
     """Add messages to a thread."""
-    return await storage.post_thread_messages(opengpts_user_id, tid, payload.messages)
+    thread = await storage.get_thread(user["user_id"], tid)
```

Comment:
same question as above

Reply:
This is also like `storage.get_thread_messages` - not validating ownership of the thread.